main/main.c

```diff
@@ -105,6 +107,25 @@ rte_strdup(const char *type, const char *s)
 	return strcpy(res, s);
 }
 
+int
+rte_log_ratelimit(uint32_t level, uint32_t logtype, const char *format, ...)
+{
+	va_list ap;
+	int ret = 0;
+	struct log_ratelimit_state *lrs =
+		&log_ratelimit_states[rte_lcore_id()];
+
+	va_start(ap, format);
+
+	if (level < lrs->ratelimit_level)
+		ret = rte_vlog(level, logtype, format, ap);
+	else if (log_ratelimit_allow(lrs))
```

Comment:
I'm confused by these `if` statements. Can you explain the behavior you intend to have?

Reply:
The first ``if`` statement means that the log level specified  has higher priority than the ratelimit log level, so we won't put any ratelimit on these logs.

The second ``if`` statement means that even we put ratelimit on the current log, however, the ratelimit algorithm **allows** to output a log entry, then we go ahead and call a ``rte_vlog()`` to log.

Reply:
That means that, if Gatekeeper starts to massively generate high priority log entries, the rate limit won't help to keep the log under a reasonable rate. My suggestion is to not log if `level > lrs->ratelimit_level` as DPDK already does, and to test `log_ratelimit_allow(lrs)` otherwise.

Reply:
Moreover, if we call `rte_log_get_level()` to get the log level, we can discard field `lrs->ratelimit_level` and avoid the trouble of synchronizing both level variables: `lrs->ratelimit_level` and its DPDK equivalent.